src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -547,7 +547,9 @@ def parse_error_handler(parse_error):
       all_tests_info = self.parse_test_info(junitxml_path, parse_error_handler, ['file', 'name'])
       for test_name, test_info in all_tests_info.items():
         test_target = self._get_target_from_test(test_info, targets)
-        self.report_test_info(self.options_scope, test_target, test_name, test_info)
+        for test_info_key, test_info_val in test_info.items():
+          key_list = [test_name, test_info_key]
+          self.report_test_info(self.options_scope, test_target, key_list, test_info_val)
```

Comment:
Given that the only user of this API is now using the "key list" API, would it be reasonable to remove the support for passing in and merging dicts? I know you spent some time on that, but this seems less error prone, etc. Up to you.